doc/explanation/opam-integration.rst

```diff
@@ -0,0 +1,115 @@
+- ``--profile release``: set the :term:`build profile` to ``release``. In particular, this ensures that warnings are not fatal.
+- ``--ignore-promoted-rules``: silently ignores all rules with ``(mode promote)``.
+- ``--default-target @install``: make sure that ``dune build`` with no target argument builds ``@install``, not ``@@default`` (this is not used in the opam integration since an explicit target is passed)
+- ``--no-config``: do not load the configuration file in the user's home directory.
+- ``--always-show-command-line``: this ensures that the programs executed by Dune end up in the opam logs.
+- ``--promote-install-files``: this ensures that ``*.install`` files are present in the source tree after the build.
+- ``--require-dune-project-file``: fail if ``dune-project`` is not present. In some previous Dune versions, ``dune-project`` could be generated when it is not present. This is not desirable with opam since the version the package has been prepared with is not known.
+
+The Targets we're Building
+--------------------------
+
+The targets are specified as:
+
+.. code::
+
+  "@install"
+  "@runtest" {with-test}
+  "@doc" {with-doc}
+
+The ``{with- }`` syntax is an opam filter. It means that the string before is
+present or not depending on the opam variable. These variables, in turn, are set depending on the opam configuration.
+
+Concretely, in the next table, if the opam command on the left is executed, the dune target on the right will be built:
+
+.. list-table::
+   :header-rows: 1
+
+   * - opam command
+     - Dune target
+   * - ``opam install pkg``
+     - ``@install``
+   * - ``opam install pkg --with-test``
+     - ``@install @runtest``
+   * - ``opam install pkg --with-test --with-doc``
+     - ``@install @runtest @doc``
+
+This filtering mechanism is also used to declare dependencies.
+If a package is using ``lwt`` and ``alcotest``, but the latter only in its test
+suite, its ``depends:`` field is:
+
+.. code::
+
+   "lwt"
+   "alcotest" {with-test}
+
+This is expanded to just ``"lwt"`` in ``opam install pkg``, but to ``"lwt"
+"alcotest"`` in ``opam install pkg --with-test``.
+
+The meaning of these :term:`aliases <alias>` is the following:
+
+- ``@install`` depends on all the ``*.install`` files in the project. In turn, these depend on all the installable files (libraries and executables with a public name, and files that are manually installed through ``(install)`` stanzas).
+- ``@runtest`` is the alias to which all tests are attached, including ``(test)`` stanzas. ``dune build @runtest`` is equivalent to ``dune runtest``.
+- ``@doc`` executes odoc to create HTML docs under ``_build``.
+
+What Opam Expects From Dune
+---------------------------
+
+Given this ``build:`` lines and the fact that there is no ``install:`` line,
+what happens is the following:
+
+- opam executes ``dune subst``, if the package is being pinned.
+- opam executes the build instruction, usually just ``dune build -p pkg @install``
```

Comment:
```suggestion
- Opam executes ``dune subst``, if the package is being pinned.
- Opam executes the build instruction, usually just ``dune build -p pkg @install``
```
Only capitalized here because it begins a sentence, even if that sentence is a bullet point.